src/Symfony/Component/Messenger/Bridge/Kafka/Transport/KafkaReceiver.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Messenger\Bridge\Kafka\Transport;
+
+use Psr\Log\LoggerInterface;
+use RdKafka\KafkaConsumer;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+/**
+ * @author Konstantin Scheumann <konstantin@konstantin.codes>
+ */
+class KafkaReceiver implements ReceiverInterface
+{
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var SerializerInterface */
+    private $serializer;
+
+    /** @var RdKafkaFactory */
+    private $rdKafkaFactory;
+
+    /** @var KafkaReceiverProperties */
+    private $properties;
+
+    /** @var KafkaConsumer */
+    private $consumer;
+
+    /** @var bool */
+    private $subscribed;
+
+    public function __construct(
+        LoggerInterface $logger,
+        SerializerInterface $serializer,
+        RdKafkaFactory $rdKafkaFactory,
+        KafkaReceiverProperties $properties
+    ) {
+        $this->logger = $logger;
+        $this->serializer = $serializer;
+        $this->rdKafkaFactory = $rdKafkaFactory;
+        $this->properties = $properties;
+
+        $this->subscribed = false;
+    }
+
+    public function get(): iterable
+    {
+        $message = $this->getSubscribedConsumer()->consume($this->properties->getReceiveTimeoutMs());
+
+        switch ($message->err) {
+            case \RD_KAFKA_RESP_ERR_NO_ERROR:
+                $this->logger->info(sprintf(
+                    'Kafka: Message %s %s %s received ',
+                    $message->topic_name,
+                    $message->partition,
+                    $message->offset
+                ));
+
+                $envelope = $this->serializer->decode([
+                    'body' => $message->payload,
+                    'headers' => $message->headers,
+                    'key' => $message->key,
+                    'partition' => $message->partition,
+                    'offset' => $message->offset,
+                    'timestamp' => $message->timestamp,
+                ]);
+
+                return [$envelope->with(new KafkaMessageStamp($message))];
+            case \RD_KAFKA_RESP_ERR__PARTITION_EOF:
+                $this->logger->info('Kafka: Partition EOF reached. Waiting for next message ...');
+                break;
+            case \RD_KAFKA_RESP_ERR__TIMED_OUT:
+                $this->logger->debug('Kafka: Consumer timeout.');
+                break;
+            case \RD_KAFKA_RESP_ERR__TRANSPORT:
+                $this->logger->debug('Kafka: Broker transport failure.');
```

Comment:
It's only a recoverable temporary glitch. The consumer can keep running.

Reply:
Same as above unfortunately - usually consumer can recover from this if connection can be reestablished. In this case I'd move this up to an `error` level, because if this happens then something wrong is happening with the broker / communication, but as said - consumer can recover from this.

Reply:
As far as I know, RD_KAFKA_RESP_ERR__TRANSPORT may be triggered by the broker itself when cleaning unused connections… This kind of "error" may appear every 10 minutes, so it’s not really an "error". The consumer is supposed to recover. By the way, it may lead to debug messages… So personnaly, I had to set 'log.connection.close' option to 'false'.
It was on a project few years ago, so I don’t know if everything is still the same. But if it does, I hope this can help…